Homework7/FightLogic.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Homework7
+{
+    class FightLogic
+    {
+        public static Unit GenerateFighter()
+        {
+            int randomIndex = Randomizer.RandomInt(0, 3);
+            Unit unit;
+
+            switch (randomIndex)
+                {
+                case 0:
+                    unit = new Warrior();
+                        break;
+
+                case 1:
+                    unit = new Archer();
+                    break;
+
+                default:
+                    unit = new Wizard();
+                    break;
+            }
+
+            return unit;
+        }
+
+        public static Unit GenerateFighter(FighterType? fighterType)
```

Comment:
In any case you can combine this methods

Reply:
fixed